src/arrays/programs_involving_multidimensional_data/hard/q6/MainTest.java

```diff
@@ -0,0 +1,133 @@
+
+public class MainTest extends BaseRandomTest {
+    // Java
+
+    public static final int n = 5;
+    public static final int m = 5;
+    public static final int up = 100;
+    public static final int down = 0;
+
+    public Clause[] testSentence() {
+        Clause[] c = new Clause[2 * n * m + n + 6];
+        int t = 0;
+        c[t++] = new StringLiteral("Generated array:");
+        c[t++] = new Optional(new StringLiteral(" "));
+        c[t++] = new NewLine();
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < m; j++) {
+                c[t++] = new RandomInteger(down, up, i + " " + j);
+                if (j != n - 1)
+                    c[t++] = new StringLiteral(" ");
+                else
+                    c[t++] = new Optional(new StringLiteral(" "));
+            }
+            c[t++] = new NewLine();
+        }
+        c[t++] = new StringLiteral("Maximum value is: ");
+        c[t++] = new IntegerLiteral("max");
+        c[t] = new Optional(new NewLine());
+
+        return c;
+    }
+
+    public void runMain() {
+        MaxVal.main(new String[0]);
+    }
+
+    @RepeatedTest(10)
+    public void printsCorrectOutput() {
+        int max = -1;
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < m; j++) {
+                int a = Integer.parseInt(getItemByName(i + " " + j));
+                if (max < a)
+                    max = a;
+            }
+        }
+        int out = Integer.parseInt(getItemByName("max"));
+        CustomAssertions._assertEquals(max, out, "Your program does not correctly find the maximum value of the 2D array.");
+    }
+
+    public static int[][] gen(int n, int m){
+        int[][] t = new int[n][m];
+        for (int i = 0; i < n; i++)
+            t[i] = generateRandomArray(down, up, m);
+        return  t;
+    }
+
+    static Stream<Arguments> inputProvider() {
+        int[][] t = gen(10, 15);
+        int max = -1;
+        for (int i = 0; i < 10; i++)
+            for (int j = 0; j < 15; j++)
+                max = Math.max(max, t[i][j]);
+        return Stream.of(
+                Arguments.of(
+                        new int[][]{
+                                {1, 2, 3},
+                                {4, 5, 4},
+                                {3, 2, 1}
+                        },
+                        5
+                ),
+                Arguments.of(
+                        new int[][]{
+                                {1, 2, 3},
+                                {4, 5, 6},
+                                {7, 8, 9}
+                        },
+                        9
+                ),
+                Arguments.of(
+                        new int[][]{
+                                {12, 11, 10, 9},
+                                {8, 7, 6, 5},
+                                {4, 3, 2, 1}
+                        },
+                        12
+                ),
+                Arguments.of(
```

Comment:
And this case are effectively the same case, so you don't need to test them both.

Reply:
It would be better to have a test case with larger numbers mixed in and put the largest value in a more obscure spot (like not the first, last, or middle spot.